### PR TITLE
add `FunctionModule.countAll(...)`.

### LINKED ISSUE #250: add `count(*)` & `count(t1.*)` support.
Currently the following syntax is not supported:

```sql
select count(*) as count from t1;
select count(t1.*) as count from t1 inner join t2 on t1.id = t2.t1_id;
```

This should probably be added as `FunctionModule.countAll(table?)`.

### PR DESCRIPTION
**Pull request by igalklebanov**
Closes #250.

- [x] implement.
- [x] unit test.
- [x] typings test.
- [x] document.

---

Also went over `FunctionModule`'s docs.

### CODE REVIEW

**Comment by koskimas** [comment]
Awesomness!

File: src/query-builder/function-module.ts [hunk 1]
```diff
@@ -198,11 +285,112 @@ export class FunctionModule<DB, TB extends keyof DB> {
...
+   * see {@link count}.
+   *
+   * For additional functionality such as filtering and window functions, refer
+   * to {@link AggregateFunctionBuilder}. An instance of this builder is returned
+   * when calling this function.
+   *
+   * ### Examples
+   *
+   * ```ts
+   * const { count } = db.fn
```
Comment (zmilonas) [inline src/query-builder/function-module.ts:304]: I think you meant `countAll` here otherwise it might not work 😅

Comment (igalklebanov) [reply src/query-builder/function-module.ts:304]: @zmilonas Nice catch! Wanna submit a PR fixing this?

Comment (zmilonas) [reply src/query-builder/function-module.ts:304]: Sure thing! Here you go: https://github.com/koskimas/kysely/pull/303

**Pull request merged**